File: CS.DotNetCore.LoadTest/src/CS.DotNetCore.LoadTest.WebApp/Data/Memory/IdentityMemoryDAO.cs
namespace CS.DotNetCore.LoadTest.WebApp.Data.Memory
{
    using Business;
    using System;
    using System.Collections.Concurrent;

    internal class IdentityMemoryDAO : IIdentityDAO
    {
        private static readonly ConcurrentDictionary<string, Identity> _dataBase = new ConcurrentDictionary<string, Identity>();

        public void Insert(Identity identity)
        {
            if (!_dataBase.TryAdd(identity.IdentityName, identity))
            {
                throw new Exception(string.Concat("insertion for identity \"", identity.IdentityName, "\" fails"));
            }
        }

        public int DeleteAll()
        {
            lock (_dataBase)
            {
                var affected = _dataBase.Count;
                _dataBase.Clear();

                return affected;
            }
        }
    }
}

File: DotNetCorePracticalView/CS.NodeJS.LoadTest.WebApp/webapp/data/memory/identityMemoryDAO.js
module.exports = {

  _database: {},

  insert: (identity) => {

    let persisted = module.exports._database[identity.identityName];

    if (persisted) {

      throw new Error('insertion for identity "' + identity.identityName + '" fails');
    }

    module.exports._database[identity.identityName] = identity;
  },

  deleteAll: () => {

    let affected = Object.keys(module.exports._database).length;
    module.exports._database = {};
    return affected;
  }
}